tools/examples_with_valgrind.py

```diff
@@ -0,0 +1,183 @@
+import os
+import subprocess
+import time
+from collections import defaultdict
+import shlex
+
```

Comment:
import shlex will probably fail if we don't add this via pip-install, etc.
do we really need it?

Reply:
I changed it to not use shlex.